src/els_indexer.erl

```diff
@@ -67,20 +64,32 @@ index_file(Path, SyncAsync) ->
   try_index_file(Path, SyncAsync),
   {ok, els_uri:uri(Path)}.
 
--spec index(els_document:document()) -> ok.
-index(Document) ->
-  Uri    = els_document:uri(Document),
-  ok     = els_db:store(documents, Uri, Document),
-  Module = els_uri:module(Uri),
-  ok = els_db:store(modules, Module, Uri),
-  Specs  = els_document:points_of_interest(Document, [spec]),
-  [els_db:store(signatures, {Module, F, A}, Tree) ||
-    #{id := {F, A}, data := Tree} <- Specs],
-  Kinds = [application, implicit_fun],
-  POIs  = els_document:points_of_interest(Document, Kinds),
-  purge_uri_references(Uri),
-  [register_reference(Uri, POI) || POI <- POIs],
-  ok.
+%% TODO: Avoid nested transactions
+-spec index(els_dt_document:item()) -> ok.
+index(#{uri := Uri, text := Text} = Document) ->
+  MD5 = erlang:md5(Text),
+  case els_dt_document:lookup(Uri) of
+    {ok, [#{md5 := MD5}]} ->
+      %% The module is already indexed, no action needed.
+      ok;
+    _ ->
+      F = fun() ->
```

Comment:
For readability, would it be possible to move this out into it's own function that takes arguments?

Reply:
Yes, but I would leave this refactoring out of this PR. The indexer module needs some love.